src/BaseHostPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Allow to modify the the schema and the host of an existing request.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class BaseHostPlugin implements Plugin
```

Comment:
Base comes from "BaseUriPlugin". But AddHostPlugin may be better. It explains better the action of the plugin.